algebra/tests/matrix/matrix_multiplication.c

```diff
@@ -0,0 +1,461 @@
+/*
+ * Phoenix-Pilot
+ *
+ * Unit tests for matrix library
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include <unity_fixture.h>
+
+#include <matrix.h>
+
+#include "tools.h"
+#include "buffs.h"
+
+
+matrix_t M1, M2, M3, Exp;
+
+
+/* ##############################################################################
+ * ----------------------        matrix_prod tests       ------------------------
+ * ############################################################################## */
+
+
+TEST_GROUP(group_matrix_prod);
+
+
+TEST_SETUP(group_matrix_prod)
+{
+	M1.data = NULL;
+	M2.data = NULL;
+	M3.data = NULL;
+	Exp.data = NULL;
+}
+
+
+TEST_TEAR_DOWN(group_matrix_prod)
+{
+	matrix_bufFree(&M1);
+	matrix_bufFree(&M2);
+	matrix_bufFree(&M3);
+	matrix_bufFree(&Exp);
+}
+
+
+TEST(group_matrix_prod, matrix_prod_std)
+{
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&M1, buffs_rowsC, buffs_colsC, buffs_C, buffs_colsC * buffs_rowsC));
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&M2, buffs_rowsD, buffs_colsD, buffs_D, buffs_colsD * buffs_rowsD));
+	TEST_ASSERT_EQUAL_INT(BUF_ALLOC_OK,
+		algebraTests_createAndFill(&Exp, buffs_rowsCtimesD, buffs_colsCtimesD, buffs_CtimesD, buffs_colsCtimesD * buffs_rowsCtimesD));
```

Comment:
There is no need to allocate memory for Exp. You can just localy declare that matrix as const
```suggestion
const matrix_t Exp = { data = buffs_CtimesD, .rows = buffs_rowsCtimesD, .cols = buffs_colsCtimesD, .transposed = 0 };
```
If this is suitable then it applies in more tests in that file.